packages/workflows/src/handlers/inventory/create-inventory-items.ts

```diff
@@ -28,47 +24,27 @@ export async function createInventoryItems({
     return void 0
   }
 
-  const variants = data.products.reduce(
-    (
-      acc: ProductTypes.ProductVariantDTO[],
-      product: ProductTypes.ProductDTO
-    ) => {
-      return acc.concat(product.variants)
-    },
-    []
-  )
-
   const result = await Promise.all(
-    variants.map(async (variant) => {
-      if (!variant.manage_inventory) {
-        return
-      }
-
-      const inventoryItem = await inventoryService!.createInventoryItem(
-        {
-          sku: variant.sku!,
-          origin_country: variant.origin_country!,
-          hs_code: variant.hs_code!,
-          mid_code: variant.mid_code!,
-          material: variant.material!,
-          weight: variant.weight!,
-          length: variant.length!,
-          height: variant.height!,
-          width: variant.width!,
-        },
-        {
-          transactionManager: (context.transactionManager ??
-            context.manager) as any,
-        }
-      )
-
-      return { variant, inventoryItem }
+    data.inventoryItems.map(async (item) => {
+      const inventoryItem = await inventoryService!.createInventoryItem({
+        sku: item.sku!,
+        origin_country: item.origin_country!,
+        hs_code: item.hs_code!,
+        mid_code: item.mid_code!,
+        material: item.material!,
+        weight: item.weight!,
+        length: item.length!,
+        height: item.height!,
+        width: item.width!,
+      })
+
+      return { tag: item._associationTag ?? inventoryItem.id, inventoryItem }
```

Comment:
comment: I would just argue on the name tag, I don't have an idea to what would be better but I fear that just reading the code we might be confusing the user between tag and tags that are associated to a product wdyt?

Reply:
Maybe we should change the input type a bit so it ends up looking like this:

```
{
  creationData: [this is where the inventory item data goes],
  workflowMetadata: { variantToAssociate: [variant id goes here] }
}
```

I have something close to it here https://github.com/medusajs/medusa/pull/5313/files#diff-cceb84f8bbe8137029bcd1d8b67e5253dea1701bfd04d710e1f26bff92517e89R13

If okay with you I'd prefer we merge this and then I can solve it in the other PR?

Reply:
Yes thats ok 👌